src/offset.js

```diff
@@ -57,10 +57,10 @@ jQuery.offset = {
 		}
 
 		if ( options.top != null ) {
-			props.top = ( options.top - curOffset.top ) + curTop;
+			props.top = ( options.top - ( curOffset ? curOffset.top : 0 ) ) + curTop;
 		}
 		if ( options.left != null ) {
-			props.left = ( options.left - curOffset.left ) + curLeft;
+			props.left = ( options.left - ( curOffset ? curOffset.left : 0 ) ) + curLeft;
```

Comment:
Yes, I thought about it and decided to not do that because it will create extra Object memory. Which is absolutely slower than simple check.